File: src/main/kotlin/com/githubyss/mobile/common/kit/binding/binding_adapter/ImageViewBindingAdapter.java
// package com.githubyss.mobile.common.kit.mvvm.bindingadapter;
//
// import android.widget.ImageView;
//
// import com.githubyss.mobile.common.kit.glide.GlideUtils;
//
// import androidx.databinding.BindingAdapter;
//
//
// /**
//  * ImageViewBindingAdapter
//  *
//  * @author Ace Yan
//  * @github githubyss
//  * @createdTime 2022/05/19 11:26:05
//  */
// public class ImageViewBindingAdapter {
//
//     @BindingAdapter({"path"})
//     public static void loadImage(ImageView view, Object path) {
//         GlideUtils.INSTANCE.loadImage(view, view, path, null, null);
//     }
//
//     @BindingAdapter(value = {"path", "placeholder", "error"}, requireAll = false)
//     public static void loadImage(ImageView view, Object path, Object placeholder, Object error) {
//         GlideUtils.INSTANCE.loadImage(view, view, path, placeholder, error);
//     }
// }

File: src/main/kotlin/com/githubyss/mobile/common/kit/binding/binding_adapter/ImageViewBindingAdapter.kt
package com.githubyss.mobile.common.kit.view_binding.binding_adapter

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.githubyss.mobile.common.kit.glide.GlideUtils


@BindingAdapter(value = ["path", "placeholder", "error"], requireAll = false)
fun ImageView.loadImage(path: Any?, placeholder: Any?, error: Any?) {
    GlideUtils.loadImage(this, this, path, placeholder, error)
}
